app/src/main/java/com/example/weather/MainActivity.java

```diff
@@ -2,13 +2,192 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.net.Uri;
 import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import javax.net.ssl.HttpsURLConnection;
 
 public class MainActivity extends AppCompatActivity {
 
+    private TextView widgetViewWeather;
+    private EditText widgetEditCity;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        widgetViewWeather = findViewById(R.id.msg_temperature);
+        widgetEditCity = findViewById(R.id.msg_city);
+    }
+
+    public void onClickByCity(View view) throws MalformedURLException {
+        Log.d("myTag", "OnClick start");
+
+        String city = widgetEditCity.getText().toString();
+        if (!checkCityString(city)) {
+            msgOnWrongCity();
+            return;
+        }
+
+        URL request = formatRequestWithCity(city);
+        Log.d("myTag", String.valueOf(request));
+
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                Log.d("myTag", "run started");
+                String respond = null;
+                try {
+                    respond = doRequest(request);
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+                if (respond != null) {
+                    Log.d("myTag", "len:" + respond.length());
+                } else {
+                    Log.d("myTag", "respond == null");
+                    widgetViewWeather.post(new Runnable() {
+                        @Override
+                        public void run() {
+                            widgetViewWeather.setText(getText(R.string.error_wrong_request));
+                        }
+                    });
+
+                    return;
+                }
+
+                String formattedOutput = parseTemperature(respond);
+                if (formattedOutput == null) {
+                    Log.d("myTag", "formattedOutput == null");
+                    widgetViewWeather.post(new Runnable() {
+                        @Override
+                        public void run() {
+                            widgetViewWeather.setText(getText(R.string.error_wrong_city));
+                        }
+                    });
+
+                    return;
+                }
+
+                String newMessage = getText(R.string.default_message) + " " + formattedOutput + " \u00B0C";
```

Comment:
Always think about good names for variables.
```suggestion
                String temperatureMessage = getText(R.string.default_message) + " " + formattedOutput + " \u00B0C";
```

Reply:
use placeholders
https://developer.android.com/guide/topics/resources/string-resource

Reply:
getText(R.string.default_message, temperature)